File: python/day1/main.py
def get_number(line: str):
    for char in line:
        if char.isdigit():
            first_digit = int(char)
            break
    for char in line[::-1]:
        if char.isdigit():
            second_digit = int(char)
            break
    number = first_digit * 10 + second_digit
    return number

    


sum = 0
with open("input.txt") as f:
    for line in f:
        sum += get_number(line)

print(sum)

File: rust/day1/src/main.rs
fn read_input() -> String {
    std::fs::read_to_string("input.txt").unwrap()
}

fn get_first_digit(text: &str) -> Option<char> {
    for char in text.chars() {
        if char.is_numeric() {
            return Some(char.clone());
        }
    }
    return None;
}

fn get_last_digit(text: &str) -> Option<char> {
    for char in text.chars().rev() {
        if char.is_numeric() {
            return Some(char.clone());
        }
    }
    return None;
}

fn main() {
    let mut sum = 0;

    let input = read_input();
    for line in input.lines() {
        let first_digit = get_first_digit(line).unwrap();
        let last_digit = get_last_digit(line).unwrap();
        let new_number: i32 = format!("{}{}", first_digit, last_digit).parse::<i32>().unwrap();
        println!("{new_number}");
        sum += new_number;
    }
    println!("{sum}");
}
